packages/lib-utils/src/app/AppInitSDK.tsx

```diff
@@ -0,0 +1,51 @@
+import { consoleLogger } from '@monorepo/common';
+import * as React from 'react';
+import { Provider } from 'react-redux';
+import type { UtilsConfig } from '../config';
+import { setUtilsConfig } from '../config';
+import type { InitAPIDiscovery } from '../types/api-discovery';
+import { initAPIDiscovery } from './api-discovery';
+import { useReduxStore } from './redux';
+
+type AppInitSDKProps = React.PropsWithChildren<{
+  configurations: {
+    apiDiscovery?: InitAPIDiscovery;
+    appFetch: UtilsConfig['appFetch'];
+  };
+}>;
+
+/**
+ * Component for providing store access to the SDK.
+ * Add this at app-level to make use of app's redux store and pass configurations prop needed to initialize the app, preferred to have it under Provider.
+ * It checks for store instance if present or not.
+ * If the store is there then the reference is persisted to be used in SDK else it creates a new store and passes it to the children with the provider
+ * @component AppInitSDK
+ * @example
+ * ```ts
+ * return (
+ *  <Provider store={store}>
+ *   <AppInitSDK configurations={{ appFetch, apiDiscovery }}>
+ *      <CustomApp />
+ *      ...
+ *   </AppInitSDK>
+ *  </Provider>
+ * )
+ * ```
```

Comment:
🤔 
I wonder if we should add an example without Redux too since that's what the `useReduxStore` hook is trying to determine as well.

Maybe not needed for now. But we should delete the `...` as we only allow 1 child:

```suggestion
 * @example
 * ```ts
 * return (
 *  <Provider store={store}>
 *   <AppInitSDK configurations={{ appFetch, apiDiscovery }}>
 *      <App />
 *   </AppInitSDK>
 *  </Provider>
 * )
 * ```
```

Reply:
Agreed on adding another example without redux. I would like to add that as a follow up to confirm everything behaves as intended in the SDK.

Reply:
Ticket tracking the work: https://issues.redhat.com/browse/HAC-759